File: Source/AlphaTab/Rendering/Effects/PickStrokeEffectInfo.cs
using AlphaTab.Model;
using AlphaTab.Rendering.Glyphs;

namespace AlphaTab.Rendering.Effects
{
    class PickStrokeEffectInfo : IEffectBarRendererInfo
    {
        public string EffectId { get { return "pick-stroke"; } }
        public bool HideOnMultiTrack { get { return false; } }
        public bool CanShareBand { get { return true; } }
        public EffectBarGlyphSizing SizingMode { get { return EffectBarGlyphSizing.SingleOnBeat; } }

        public bool ShouldCreateGlyph(Settings settings, Beat beat)
        {
            return beat.PickStroke != PickStroke.None;
        }

        public EffectGlyph CreateNewGlyph(BarRendererBase renderer, Beat beat)
        {
            return new PickStrokeGlyph(0, 0, beat.PickStroke);
        }

        public bool CanExpand(Beat @from, Beat to)
        {
            return true;
        }
    }
}

File: packages/alphatab/src/rendering/effects/PickStrokeEffectInfo.ts
import type { Beat } from '@coderline/alphatab/model/Beat';
import { PickStroke } from '@coderline/alphatab/model/PickStroke';
import type { BarRendererBase } from '@coderline/alphatab/rendering/BarRendererBase';
import { EffectBarGlyphSizing } from '@coderline/alphatab/rendering/EffectBarGlyphSizing';
import type { EffectGlyph } from '@coderline/alphatab/rendering/glyphs/EffectGlyph';
import { PickStrokeGlyph } from '@coderline/alphatab/rendering/glyphs/PickStrokeGlyph';
import { EffectBarRendererInfo } from '@coderline/alphatab/rendering/EffectBarRendererInfo';
import type { Settings } from '@coderline/alphatab/Settings';
import { NotationElement } from '@coderline/alphatab/NotationSettings';

/**
 * @internal
 */
export class PickStrokeEffectInfo extends EffectBarRendererInfo {
    public get notationElement(): NotationElement {
        return NotationElement.EffectPickStroke;
    }

    public get hideOnMultiTrack(): boolean {
        return false;
    }

    public get canShareBand(): boolean {
        return true;
    }

    public get sizingMode(): EffectBarGlyphSizing {
        return EffectBarGlyphSizing.SingleOnBeat;
    }

    public shouldCreateGlyph(_settings: Settings, beat: Beat): boolean {
        return beat.pickStroke !== PickStroke.None;
    }

    public createNewGlyph(_renderer: BarRendererBase, beat: Beat): EffectGlyph {
        return new PickStrokeGlyph(0, 0, beat.pickStroke);
    }

    public canExpand(_from: Beat, _to: Beat): boolean {
        return true;
    }
}
